### PR TITLE
[Bugfix] Fix guidance backend for Qwen models

### PR DESCRIPTION
**Pull request by benchislett**
The smaller of the Qwen models (such as `Qwen/Qwen2.5-Coder-3B-Instruct` for instance) have padding on their vocab size, such that the tokenizer will report a `vocab_size` which is smaller than the logits by 256 or 128.

This is an error for llguidance, which generates masks of the smaller size and therefore does not mask all bits. This leads to a guidance failure, producing no output.

The optional second argument overrides the vocab size to be the correct one.

### CODE REVIEW

**Review by mmoskal — APPROVED** [review]
Perfect, thank you!

**Review by russellb — APPROVED** [review]

**Review by mgoin — APPROVED** [review]
